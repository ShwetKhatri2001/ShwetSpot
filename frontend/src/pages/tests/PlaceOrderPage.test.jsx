import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import PlaceOrderPage from '../PlaceOrderPage';
import store from '../../redux/store';

describe('PlaceOrderPage Component', () => {
  const MockPlaceOrderPage = (
    <MemoryRouter>
      <Provider store={store}>
        <PlaceOrderPage />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockPlaceOrderPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
