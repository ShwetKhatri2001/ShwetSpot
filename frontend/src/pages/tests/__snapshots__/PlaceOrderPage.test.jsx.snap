// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceOrderPage Component should match the snapshot 1`] = `
Array [
  <div
    className="justify-content-center mb-4 nav"
    onKeyDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link active"
        data-rb-event-key="/login"
        href="/login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Sign In
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/shipping"
        href="/shipping"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Shipping
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/payment"
        href="/payment"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Payment
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/placeorder"
        href="/placeorder"
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "backgroundColor": "#f8f8f8",
          }
        }
      >
        Place Order
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <h2>
            Shipping
          </h2>
          <p>
            <strong>
              Address:
            </strong>
            , 
             
            ,
             
          </p>
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <h2>
            Payment Method
          </h2>
          <strong>
            Method: 
          </strong>
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <h2>
            Order Items
          </h2>
          <div
            className="fade alert alert-info show"
            role="alert"
          >
            Your cart is empty
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="card"
      >
        <div
          className="list-group list-group-flush"
          onKeyDown={[Function]}
        >
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <h2>
              Order Summary
            </h2>
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                Items
              </div>
              <div
                className="col"
              >
                0.00
                 credits
              </div>
            </div>
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                Shipping
              </div>
              <div
                className="col"
              >
                100.00
                 credits
              </div>
            </div>
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                Tax
              </div>
              <div
                className="col"
              >
                0.00
                 credits
              </div>
            </div>
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                Total
              </div>
              <div
                className="col"
              >
                100.00
                 credits
              </div>
            </div>
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="btn-block btn btn-success"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Place Order
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
